gridded_etl_tools/utils/zarr_methods.py

```diff
@@ -1009,14 +1065,54 @@ def is_append_contiguous(self, original_dataset: xr.Dataset, append_times: list[
         bool
             whether the original and appending datasets form a contiguous time axis
         """
-        # Use time step of the original dataset to get expected time delta. This could be improved by using Attributes.temporal_resolution
-        # if it were edited to return a numpy.timedelta64 (what if previous dataset has only one time step, for example).
+        # Check if first time in append times is one time step ahead of the last time in the original dataset
         last_time_in_original = original_dataset[self.time_dim].values[-1]
-        expected_delta = last_time_in_original - original_dataset[self.time_dim].values[-2]
+        self.is_time_contiguous(append_times[0], last_time_in_original)
 
-        # Check if first time in append times is one time step ahead of the last time in the original dataset
-        if append_times[0] - last_time_in_original != expected_delta:
-            return False
+        # Check if all times to be appended are contiguous with the prior time step in the append time range
+        self.are_times_contiguous(append_times)
+
+    def are_times_contiguous(self, times: tuple[datetime.datetime]) -> bool:
+        """
+        Convenience method to run `is_time_contiguous` in a loop over an ascending ordered list of input times,
+        since this is a regular pattern
+
+        Parameters
+        ----------
+        times
+            A datetime.datetime object representing the timestamp being checked
+
+        Returns
+        -------
+        bool | None
```

Comment:
I don't think this function could return `None` and it probably shouldn't anyway